code/game/objects/items/shields.dm

```diff
@@ -36,13 +36,28 @@
 		if (obj_integrity <= attackforce)
 			var/turf/T = get_turf(owner)
 			T.visible_message("<span class='warning'>[hitby] destroys [src]!</span>")
+			obj_integrity = 1
 			shatter(owner)
 			return FALSE
 		take_damage(attackforce * ((100-(block_power))/100))
 		return TRUE
 	else
 		return ..()
 
+/obj/item/shield/attackby(obj/item/weldingtool/W, mob/living/user, params)

```

Comment:
if(!istype(W))
    return

You don't need a welder to repair it
![image](https://user-images.githubusercontent.com/26465327/81493790-4301d080-929b-11ea-8b92-4c7062fa379b.png)